src/assignments/tech-jobs-mvc.rst

```diff
@@ -0,0 +1,443 @@
+other way of searching. This action method is called as a
+result of clicking on a link within the ``List`` view, rather than via
+submitting a form. We’ll see where these links originate when we look at the
+views. 
+
+The ``SearchController``
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Currently, the search controller contains only a single action method, ``Index``.
+It simply renders the form defined in the ``Views/Search/Index.cshtml`` template.
+
+Later in this assignment, you will receive instructions for adding a second
+action method to deal with user input and display the search results.
+
+The Views
+^^^^^^^^^
+
+Let’s turn our attention to the views.
+
+Bootstrap Classes
+~~~~~~~~~~~~~~~~~
+
+The application uses a few Bootstrap classes to style the view content and
+job tables. You won’t have to explicitly add any Bootstrap classes to your views in this assignment,
+but it’s a great way to make your sites look good with minimal work.
+
+The List Views
+~~~~~~~~~~~~~~
+
+Turn your attention to ``List/Index.cshtml``. This page displays a table of links
+broken down into several categories. Data from ``ListController.ColumnChoices`` is used to
+fill in the header row, and information stored in ``ListController.TableChoices`` generates
+the link text.
+
+The most interesting part of this template is how we generate the links:
+
+.. sourcecode:: html
+   :lineno-start: 17
+
+   @foreach (var category in ViewBag.tableChoices)
+   {
+      <td>
+         <ul>
+            @foreach (var item in category.Value)
+            {
+               <li>
+                  <a asp-controller="List" asp-action="Jobs" asp-route-column="@category.Key" asp-route-value="@item">@item</a>
+               </li>
+            }
+         </ul>
+      </td>
+   }
+
+#. ``ViewBag.tableChoices`` is the ``TableChoices`` dictionary from ``ListController``. It contains the names of
+   the ``Job`` fields as keys (``employer``, etc.). The value for each key is
+   a list of ``Employer``, ``Location``, ``CoreCompetency``, or
+   ``PositionType`` objects.
+#. In line 17, ``category`` represents one key/value pair from
+   ``TableChoices``, and in line 21, ``item`` represents one entry from the
+   stored list.
+#. We’ve seen the ASP.NET action tag helpers ``asp-controller`` and ``asp-action`` before. Here, we have a couple new helper tags,
+   ``asp-route-column`` and ``asp-route-value``. This syntax, ``asp-route-*`` allows Razor
+   to dynamically generate query parameters for our URL. So whatever string takes the place of the last portion of the tag, in our case, 
+   ``-column`` and ``-value`` become the names of the query parameters the ``ListController.Jobs`` action method uses.
+
+   .. admonition:: Tip
+
+      Try clicking on a few of these links to help grasp the new syntax. Clicking on one of these ``<a>`` elements 
+      routes the app to a page at a URL that looks like this: 
+      
+      https://localhost:5001/List/Jobs?column=employer&value=Buzzbold.
```

Comment:
A helpful addition to this tip may also be showing what the HTML looks like once the site is running.